Planet-Planet interaction

EV4g opened the pull request:
Added planet-planet interaction
Updated the attractMass function to also work between two planets, and not only a planet and a sun
Explode() can now also take a arraylist as input, which will be used in the future for planet collisions/destruction

EV4g committed:
Added Planet-Planet interactions

Planets now act based on the gravitational influence of other planets aswell, instead of solely the sun. 
Reworked the attractMass() funtion so planet-planet attraction can be calculated.

EV4g committed:
Planet-Planet roche

Added a function to calculate whether or not a planet in inside anothers' roche limit, for later use.

EV4g committed:
General update

Updated attractMass() function and removed planet collisions because they really don't work.

EV4g committed:
.normalize added

explode now explodes in a circle, not in a square.

EV4g commented:
There are still things which need to be added, but planet-planet interactions work.
A repelling force should be added, so planets dont clutter and destroy the universe.

EV4g committed:
Repell

Added a slight repelling force to avoid extreme planet clutter in high-density areas, by using explode() for example

EV4g committed:
Added explode_me 

Added an explode_me boolean, so that after the physics is done, we can loop through the <planets> and explode each planet with the explode_me set to true, without breaking reality itself.

EV4g committed:
Explode upon impact

Planets now explode upon impact with another planet by setting explode_me to true, and in another loop actually exploding them, if needed.

EV4g committed:
Improved collision-based explosion

Will only violently explode if the angle is correct, so explode() works better. Explode_me is now also base on mass and radius ratio, so a extemely small asteroid can not explode a super massive giant.

EV4g committed:
< Enable / Disable > 

Added buttons
r --> enable/disable rochelimit
f --> enable/disable planet-planet interactions and destruction

EV4g committed:
renamed Physics

renamed enablePhysics to enablePtPPhysics

EV4g commented:
Latest version is a stable version

EV4g committed:
Added buttons

Added R and F

EV4g committed:
\

\

EV4g committed:
gif nr4, planet impact

< impact / destruction >

EV4g committed:
new gif

EV4g committed:
rewording

TechnicJelle commented on Planet.pde:
```diff
@@ -10,14 +10,16 @@ class Planet {
 
   boolean onScreen = true;
-
+  boolean explode_me;
```
Please rename this variable to `explodeMe`, to comply with the naming style of the rest of the variables.

Please also move the ` = false` out of the constructor into the raw class

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -125,20 +129,43 @@ void draw() {
   if (!mousePressed && simRunning && !simHalted) {
     for (int t = 0; t < TIMESTEPS_PER_FRAME * facSimSpeedMod; t++) {
-      for (int i = planets.size() - 1; i >= 0; i--) {
+      for (int i = planets.size() - 1; i >= 0; i--) {    
```
Some trailing spaces

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -275,7 +302,39 @@ PVector attractMass(Planet p) {
+  float d = dist(p.pos, o.pos);
+  float m = o.mass * p.mass;
+
+  if (d > p.radius + o.radius) { 
```
Trailing space

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -400,7 +470,7 @@ void keyPressed() {
         Planet p = planets.get(i);
         if (dist(mouseX, mouseY, p.pos.x, p.pos.y) < p.radius) {
-          explode(p, i);
+          explode(p, i, planets);
```
Feels redundant. `planets` is a global variable

TechnicJelle commented on Planet.pde:
```diff
@@ -28,7 +30,7 @@ class Planet {
     acc.add(f);
   }
-
+      
```
Trailing spaces

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -125,20 +129,43 @@ void draw() {
+        //roche limit
+        if (dist(p.pos, sun) < 2.456 * SUN_RADIUS * pow((SUN_MASS * p.radius * p.radius * p.radius) / (p.mass * SUN_RADIUS * SUN_RADIUS * SUN_RADIUS), (1.0 / 3.0)) && enableRoche) {
+          explode(p, i, planets);
+        }         
```
Trailing spaces

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -125,20 +129,43 @@ void draw() {
+
+          //Planets explode upon impact
+          for (int z = planets.size() - 1; z >= 0; z--) {
+            if (z != i && dist(p.pos, planets.get(z).pos) < (p.radius + planets.get(z).radius) / 2 && degrees(PVector.angleBetween(planets.get(z).vel, p.vel)) > 10 && p.mass < 10 * planets.get(z).mass && p.radius < 10 * planets.get(z).radius) {            
```
Trailing spaces

TechnicJelle commented on SolarSystem.pde:
```diff
@@ -275,7 +302,39 @@ PVector attractMass(Planet p) {
+  }
+}
+
+boolean roche(Planet p, Planet M) {  
```
Trailing spaces

TechnicJelle reviewed (changes requested):
Some style changes, but the rest is looking good!

EV4g committed:
Variable naming rework

explode_me --> explodeMe
and the explodeMe boolean now gets set to false in a different place

EV4g committed:
Reworked variable naming

Removed several trailing spaces
Removed <planets> from explode()

EV4g commented:
Changes have been made
